### PR TITLE
Working getIdentity() call

### LINKED ISSUE #173: user.GetIdentity() doesn't work.
It's seriously broken - uses the gateway.

### PR DESCRIPTION
**Pull request by adrianhall**
This version of the getIdentity() call does not do any post-processing of the response from the server.

Fixes #173.

### CODE REVIEW

File: src/auth/getIdentity.js [hunk 1]
```diff
@@ -2,22 +2,45 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // ----------------------------------------------------------------------------
 var https = require('https'),
+    URL = require('url'),
```
Comment (danderson00) [inline src/auth/getIdentity.js]: No uppercase variable names

File: src/auth/getIdentity.js [hunk 2]
```diff
@@ -2,22 +2,45 @@
...
+           var responseData = '';
+           response.setEncoding('utf8');
+           response.on('data', function (chunk) {
+               responseData += chunk;
+           });
+           response.on('end', function () {
+               log.verbose('[getIdentity] Response Data: ', responseData);
+               var responseObj = JSON.parse(responseData);
+               resolve(responseObj);
+           });
```
Comment (danderson00) [inline src/auth/getIdentity.js]: Do we need the stream handling here? We should be able to get the entire response without it, like in the previous version.

Comment (adrianhall) [inline src/auth/getIdentity.js]: Ideally, we would use something like the 'request' module for handling all the back and forth.  However, I wanted to get the basics in first.  What I implemented is the "recipe" from the Node site.

File: src/auth/getIdentity.js (hunk 1 above)
Comment (adrianhall) [inline src/auth/getIdentity.js]: I accidentally included this change in the normalizeClaims pull request.  If you are happy with both, then merge this one then that one.

**Pull request merged**